packages/language-server/src/server.ts

```diff
@@ -86,6 +92,22 @@ documents.onDidChangeContent((change) => {
   });
 });
 
+documents.onDidChangeContent((change) => {
+  const { port1, port2 } = new MessageChannel();
+
+  const { document } = change;
+  const query = document.getText();
+
+  port2.on('message', (diagnostics: Diagnostic[]) => {
+    void connection.sendDiagnostics({
+      uri: document.uri,
+      diagnostics: diagnostics,
+    });
+  });
+
+  semanticAnalysisWorker.postMessage({ query, port: port1 }, [port1]);
+});
+
```

Comment:
I think you cannot have two `onDidChangeContent` because that overwrites it. In other words right now we are not surfacing syntactic errors

<img width="332" src="https://github.com/neo4j/cypher-language-support/assets/5649971/1236eb83-037a-4f3d-a2b4-f7977558c3bb">

Reply:
I merged them back as per our offline discussion